File: src/fibonacci.hs
-- Retorna o n-ésimo termo da sequência de Fibonacci

fibonacci :: Int -> Int
fibonacci_aux :: Int -> Int -> Int -> Int

-- Atribui parâmetros padrão do caso base (Dois números iniciais)
fibonacci n = fibonacci_aux n 0 1
    
fibonacci_aux n a b
    | n == 1 = b
    | otherwise = fibonacci_aux (n - 1) b (a + b)
    
main :: IO ()
main =  do
    print(fibonacci 6)
    print(fibonacci 8)

File: src/fibonacci.pl
% Retorna o n-ésimo termo da sequência de Fibonacci

fibonacci(N, Result) :- fibonacci_tail(N, 0, 1, Result).

fibonacci_tail(0, N, _, N).
fibonacci_tail(N, A, B, Result) :-
   N > 0,
   New_B is A + B,
   N1 is N - 1,
   fibonacci_tail(N1, B, New_B, Result).
    
% Consulta ---> fibonacci(6, R).
